ProjectWar/Map.cpp

```diff
@@ -349,31 +349,99 @@ void Map::checkNearEntities(Unit *unit, std::list<UnitCommand>& commands)
     }
 }
 
+void Map::updateInfoTileBuilding(Unit* unit, Point destination){
+    
+    InfoTile* tile = infoMap[unit->getPosition().x][unit->getPosition().y];
+    InfoTile* destinationTile = infoMap[destination.x][destination.y];
+    int ownerID = tile->ownerID;
+    
+    //clean tile
+    tile->text->setTextResource("");
+    tile->ownerID = -1;
+    tile->entity = NOT_DEFINED;
+    //update new tile
+    destinationTile->ownerID = ownerID;
+    destinationTile->entity = UNIT_CAPTURING;
+    destinationTile->unitID = unit->getId();
+    destinationTile->text->setTextResource(
+                                           "Unit: " +
+                                           std::to_string(destinationTile->ownerID)+
+                                           "-"+
+                                           std::to_string(destinationTile->unitID));
+    destinationTile->text->setPosition(this->getAbsolutePosition(destination));
+}
+
+void Map::updateInfoTileUnitCapturing(Unit* unit, Point destination){
+    InfoTile* tile = infoMap[unit->getPosition().x][unit->getPosition().y];
+    InfoTile* destinationTile = infoMap[destination.x][destination.y];
+    int ownerID = tile->ownerID;
+    Point position;
+    position.x = unit->getPosition().x;
+    position.y = unit->getPosition().y;
+    
+    //clean tile
+    tile->text->setTextResource("");
+    tile->ownerID = getBuilding(position)->getOwnerID();
```

Comment:
since you are using position just once i think you can just do getBuilding( unit->getPosition() )....